redis/release_lock.lua

```diff
@@ -4,9 +4,24 @@ local stored_jid = redis.pcall('get', unique_key)
 
 if stored_jid then
   if stored_jid == job_id or stored_jid == '2' then
-    redis.pcall('del', unique_key)
-    redis.pcall('hdel', 'uniquejobs', job_id)
-    return 1
+    if redis.pcall('del', unique_key) then
+      if redis.pcall('hdel', 'uniquejobs', job_id) then
+        -- All should be good
+        return 1
+      else
+        -- hdel has failed
+        return 2
+      end
+    else
+      -- del has failed, lets try hdel anyway.
+      if redis.pcall('hdel', 'uniquejobs', job_id) then
```

Comment:
didn't mean to be concise, smart or pretty or anything like that.
I am keeping functionality 100% one-for-one, I believe.
but adding more return code to help troubleshooting

because the lua scripts returns only 0/1, we don't know exactly what is happening.
with this, I have a better idea, although I dont know enough how this gem works to understand it fully.

i just turn several of my workers to be using `:until_and_while_executing` and looking at the logs, for one given job trying to pick up the lock, it looks like it is trying to steal the lock from the others... https://gist.github.com/mathieujobin/27aa191896624994d536

it does not make much sense to me.

is it possible there is some lock common to a unicity type ?

thanks